static/js/tools/stat_var/dataset_selector.tsx

```diff
@@ -0,0 +1,231 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Component to select a source and dataset.
+ */
+
+import axios from "axios";
+import React, { useEffect, useState } from "react";
+import { Card, Container, CustomInput } from "reactstrap";
+
+import { NamedPlace } from "../../shared/types";
+
+const PREFIX = "dataset-selector";
+const MAX_SUGGESTIONS = 5;
+
+interface DatasetSelectorProps {
+  // Filter StatVarHierarchy to stat vars that exist for the entities.
+  filterStatVars: (entities: NamedPlace[]) => void;
+  // Map of source name to dcid.
+  sourceMap: Record<string, string>;
+}
+
+function simplify(s: string): string {
+  return s.replace(/[^0-9a-z]/gi, "").toLowerCase();
+}
+
+export function DatasetSelector(props: DatasetSelectorProps): JSX.Element {
+  const [activeSuggestion, setActiveSuggestion] = useState(0);
+  const [datasets, setDatasets] = useState([]);
+  const [filteredSuggestions, setFilteredSuggestions] = useState(
+    Object.keys(props.sourceMap)
+  );
+  const [showSuggestions, setShowSuggestions] = useState(false);
+  const [suggestionsWidth, setSuggestionsWidth] = useState(null);
+  const [userInput, setUserInput] = useState("");
+
+  const handleResize = () => {
+    const width = document.getElementById(`${PREFIX}-ac`).offsetWidth;
+    if (width !== suggestionsWidth) {
+      setSuggestionsWidth(width);
```

Comment:
Why does this width needs to be manually calculated and updated - does CSS/flexbox not ensure the suggestions width matches the input width?

Reply:
this was mostly to keep the absolute positioning so the autocomplete dropdown would show up above the rest of the search box - I tried getting it to work with the flex instead (which does do the scaling) but wasn't sure how to get it to be visible... however I'm not very familiar with css unfortunately ;; so if you have other suggestions for what I could do, I can definitely try!

Reply:
hm, why does the autocomplete need to show up above the search box? I don't think they would need to overlap right since it goes underneath the search box? or can you take a screenshot/video of the problem you had?

Reply:
sorry eventually got it working with the flex!